### PR TITLE
Add pytorch-cuda-fix-xpu-alignment agent skill

### PR DESCRIPTION
**Pull request by laifenxiawucha**
## Summary

Add a new agent skill (`pytorch-cuda-fix-xpu-alignment`) that discovers CUDA and other backend bug fixes in `pytorch/pytorch` and validates whether the same bugs exist on XPU.

## Input / Output

**Input**: A time window (e.g., "last 1 day") and target repository (`pytorch/pytorch`).

**Output**: A local `triage_scan_<date>.md` file containing:
- Candidate list with URLs, titles, and dates
- Filter decisions (reject/pass) with reasons for each candidate
- Adapted XPU reproducer scripts for passed candidates
- Local XPU nightly validation results (confirmed / not-reproduced / unverified)
- Filter and validation statistics summary

## Workflow example
**Use**r: "Scan pytorch/pytorch for CUDA bug fixes in the last 1 day and check if they affect XPU."

The skill then:
1. **Collect** — Search `pytorch/pytorch` via GitHub MCP for recent backend bug-fix signals (issues, PRs, commits). Save a lightweight candidate list to `triage_scan_<date>.md`.
2. **Process one at a time** — For each candidate:
   - **Filter**: Read details, reject if infra-only / build-CI-only / docs-typo-only; otherwise pass
   - **Reproduce**: Extract or adapt the upstream regression test for `torch.xpu`
   - **Validate**: Run on local XPU nightly, record confirmed / not-reproduced / unverified
3. **Summarize** — Append filter statistics and validation statistics to the scan file.

## What is included

| File | Purpose |
|------|---------|
| `.github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md` | Skill definition and end-to-end workflow |
| `.github/copilot-instructions.md` | Registered in the skills index (1 row added) |

## What is missing

I also have two scripts:

- One script is used to set up a daily scheduled run: `daily_scan.sh`.
- Another script, `batch_scan.sh`, is used for batch scanning at scale, such as large-volume scans covering around 90 days.

In addition, there will be a README that explains how to configure `cron` and provides a quick start guide.

## Output Structure

Each run produces:
```
runs/<date-or-range>/
├── artifacts/
│   ├── raw_candidates.json       — deduplicated search results
│   ├── candidate_ledger.jsonl    — state-machine progress tracker
│   ├── details/<id>.json         — fetched issue/PR bodies and commit diffs
│   ├── output_<id>.log           — repro execution output
│   └── collect_env.txt           — torch environment info
├── scripts/
│   └── repro_<id>.py             — reproducer scripts
└── reports/
    └── full_scan.md              — final scan report
```

### CODE REVIEW

File: .github/skills/pytorch-cuda-xpu-triage/scripts/find_xpu_python.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+# find_xpu_python.sh
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/scripts/find_xpu_python.sh]: This file should not be included, I don't think it should be included as a skill

File: .github/skills/pytorch-cuda-xpu-triage/scripts/run_collect_env.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/scripts/run_collect_env.sh]: this should be included as a skill. You could just write in skill.md and ask the LLm to run the collect_env.  No need a separate script. It is hard to maintain

File: .github/skills/pytorch-cuda-xpu-triage/scripts/run_with_xpu_python.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/scripts/run_with_xpu_python.sh]: Similar with this file, I don't think it is needed.

File: .github/skills/pytorch-cuda-xpu-triage/scripts/update_torch_xpu_nightly.sh [hunk 4]
```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/scripts/update_torch_xpu_nightly.sh]: This file is also not needed. All you need to do is just ask the LLM to upgrade the torch with command:

```
pip3 install --pre torch torchvision torchaudio --force-reinstall --index-url https://download.pytorch.org/whl/nightly/xpu
```

File: .github/skills/pytorch-cuda-xpu-triage/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+- Produce concise repro plans and minimal Python reproducers for every qualified candidate, not just the top one.
+- Stop before filing anything unless local XPU execution confirms a bug.
+
+## When to use me
+Use this skill when you want to mine PyTorch backend semantic fixes for likely XPU parity gaps and validate them locally on an XPU torch nightly build.
+
+Interpret user focus words as a soft hint about the desired bug family, edge condition, or semantic shape unless the user explicitly says the match must be exact.
+
+## Required tools and context
+- Use the github MCP server.
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/SKILL.md]: The github MCP server currently, only installed on your local machine, so I believe that the remote should be tested. Please be aware of it

File: .github/skills/pytorch-cuda-xpu-triage/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
+---
+name: pytorch-cuda-xpu-triage
+description: Scan pytorch backend bug-fix issues, PRs, and commits with GitHub MCP, identify likely XPU-divergence bug families, and generate local reproducers for XPU nightly validation. Use when mining pytorch/pytorch for backend semantic fixes that may affect XPU parity — especially non-contiguous tensors, empty tensors, dtype promotion, reductions, advanced indexing, and masked ops.
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/SKILL.md]: This description is too long and too concrete.

File: .github/skills/pytorch-cuda-xpu-triage/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+- Search pytorch/pytorch issues and pull requests for backend-divergence bug-fix signals.
+- Pivot from issue to PR to commit and extract the minimal bug pattern.
+- Rank candidates by likelihood of backend divergence affecting XPU.
+- Produce concise repro plans and minimal Python reproducers for every qualified candidate, not just the top one.
+- Stop before filing anything unless local XPU execution confirms a bug.
+
+## When to use me
+Use this skill when you want to mine PyTorch backend semantic fixes for likely XPU parity gaps and validate them locally on an XPU torch nightly build.
+
+Interpret user focus words as a soft hint about the desired bug family, edge condition, or semantic shape unless the user explicitly says the match must be exact.
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/SKILL.md]: Do we need these lines?

File: .github/skills/pytorch-cuda-xpu-triage/SKILL.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+- The issue or PR shows CPU correctness but CUDA divergence.
+- The operator is implemented on multiple accelerators where semantic drift is plausible.
+- The candidate preserves the user's edge condition or failure shape even if the exact operator name is different.
+- The fix lives in shared frontend validation, composite fallback, common ATen/native code, or other code paths likely inherited by XPU.
+
+Deprioritize candidates dominated by:
+- NVIDIA library integration specifics without semantic implications.
+- CUDA graph, Triton, or compiler-only infrastructure.
+- Build-system, packaging, or CI-only failures.
+- Pure performance regressions without correctness impact.
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/SKILL.md]: Please include this performance regression. Do not deprioritize it.

File: .github/skills/pytorch-cuda-xpu-triage/references/local-xpu-validation-reference.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+## Minimal repro script checklist
+- imports only what is necessary
+- prints environment metadata
+- checks whether torch.xpu exists and is available
+- creates the smallest tensors that still hit the edge case
+- compares CPU and XPU behavior when possible
+- prints mismatch details, exception type, and message
+
+## Suggested environment print block
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/references/local-xpu-validation-reference.md]: As we are using collect_env.py from pytorch, so this is not needed.

**Review by Stonepia — APPROVED** [review]
Overall LGTM, we could quick enable for the init stage

**Comment by Stonepia** [comment]
Next step should land this skills. Currently, this skill only working with local OpenCode/Codex env

**Review by Stonepia — APPROVED** [review]

**Review by Stonepia — APPROVED** [review]

File: .github/skills/pytorch-cuda-xpu-triage/SKILL.md [hunk 10]
```diff
@@ -0,0 +1,147 @@
+---
+name: pytorch-cuda-xpu-triage
+description: Find recent upstream backend fixes that may expose XPU gaps and draft local repro steps. Use when an agent is asked to scan pytorch/pytorch for CUDA, XPU, or backend-divergence issues and prepare candidates for local validation.
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/SKILL.md]: Please refine this description

**Review by Stonepia — CHANGES_REQUESTED** [review]
Overall LGTM, but please refine the description for skills

**Review by ZhaoqiongZ — APPROVED** [review]

File: .github/skills/pytorch-cuda-xpu-triage/references/github-mcp-reference.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
+# GitHub Search Reference
+
+## Capabilities assumed
+The agent environment provides GitHub search, issue/PR read, commit inspection, and file content retrieval. If unavailable, stop and report what needs manual lookup.
+
+## Example queries
+```
+repo:pytorch/pytorch is:issue "incorrect result" CUDA
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/references/github-mcp-reference.md]: I still think this is too limited. You could just ask the AI to filter. We don't want to set limitation to it. Just provide one example is enough.

File: .github/skills/pytorch-cuda-xpu-triage/references/github-mcp-reference.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+3. Read linked issues or PRs if the bug scenario is unclear.
+4. Inspect only the touched operator/test files in the relevant diff.
+5. Synthesize a minimal XPU repro from the pattern.
+
+## Pitfalls
+- Not every CUDA fix is an XPU candidate.
+- Skip large compiler/infra PRs.
+- A test name alone often reveals the bug shape — avoid reading entire file diffs.
+- Skip large compiler/infra PRs.
+- A test name alone often reveals the bug shape — avoid reading entire file diffs.
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/references/github-mcp-reference.md]: these points are duplicate

File: .github/skills/pytorch-cuda-xpu-triage/references/local-xpu-validation-reference.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+```bash
+python -m pip install --pre torch --force-reinstall --index-url https://download.pytorch.org/whl/nightly/xpu
+```
+Use a dedicated virtualenv/conda env. Add `torchvision`/`torchaudio` only if the repro needs them.
+
+If the correct interpreter is unclear, ask the user.
+
+## Repro script requirements
+- Print environment metadata (`torch.__version__`, `torch.xpu.is_available()`)
+- Use the smallest tensors that hit the edge case
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/references/local-xpu-validation-reference.md]: This is hard to be implemented. I would like to delete this restriction

Comment (Stonepia) [reply .github/skills/pytorch-cuda-xpu-triage/references/local-xpu-validation-reference.md]: You could ask it just follow the original reproducer / test case

File: .github/skills/pytorch-cuda-xpu-triage/references/local-xpu-validation-reference.md [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+- Compare CPU vs XPU output
+- Print mismatch details or exception info
+
+## Confirmed bug criteria
+One of the following on XPU, when CPU semantics disagree:
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/references/local-xpu-validation-reference.md]: This part is also kind of too limited. We currently don't have a good "what kind of bug do we encountered", so I would prefer just too kind of bugs:

```
1. functionality: Like crash / wrong result / ...
2. Performance: Like there is a performance regression, etc.
```

After you have looped for some time, then collect all the data you need, we got some details of what kind of error it is

Comment (Stonepia) [reply .github/skills/pytorch-cuda-xpu-triage/references/local-xpu-validation-reference.md]: Also, "when CPU semantics disagree:" , this leaves the case when XPU directly crash, or there is no cpu output, you won't go into this logic

File: .github/skills/pytorch-cuda-fix-xpu-alignment/references/local-xpu-validation-reference.md [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+- Backward mismatch where forward is expected to work
+
+**Not** a bug by itself: tiny float noise, documented unsupported paths, or failures from an invalid repro.
+
+## Evidence to capture before escalation
+- Exact command and output
+- `python -W ignore::RuntimeWarning -m torch.utils.collect_env`
+- Full exception text or mismatch summary
+- Minimal repro script
+- Upstream issue/PR/commit links
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-fix-xpu-alignment/references/local-xpu-validation-reference.md]: duplicate

File: .github/skills/pytorch-cuda-xpu-triage/SKILL.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+Scan `pytorch/pytorch` for issues, PRs, and commits across any backend (CUDA, ROCm, CPU, etc.), generate minimal reproducers, and run them on the latest local XPU torch nightly to determine whether XPU shares the same bug.
+
+Detailed reference:
+- [references/github-mcp-reference.md](references/github-mcp-reference.md) — search patterns and narrowing workflow
+- [references/local-xpu-validation-reference.md](references/local-xpu-validation-reference.md) — environment, run commands, confirmation criteria
+
+## Workflow
+
+### Step 1: Search upstream signals
+- Search issues, PRs, and commits in `pytorch/pytorch` for backend bug-fix signals — both open and closed/merged. Sources may be CUDA-specific, ROCm-specific, or cross-backend — the key question is whether the bug pattern could also manifest on XPU.
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/SKILL.md]: You should specify how to collect. Because you are telling the AI to get. For example, you will ask it to scratch from the github and store locally, then use what kind of rule / what tools , etc.

This process need to be reproducible. Otherwise, you are totally relying on the model, so if we are going to scan ALL commits in PyTorch, if we are using a weak model, this might miss a lot of info

File: .github/skills/pytorch-cuda-xpu-triage/SKILL.md [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+- Use the time window specified by the caller (e.g., last 1 day, last 7 days, or all time). Collect **all** matching candidates in the window, not just the first few.
+- Follow links between issues, PRs, and commits to narrow down the exact bug trigger (operator, shape, dtype, edge case).
+
+### Step 2: Qualify candidates
+For each candidate, classify as **qualified** or **rejected**.
+
+Qualify when:
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/SKILL.md]: I prefer to delete that "qualified", all issues that not rejected are qualified (at the first stage). Then if there are too many issues, then filter out.

File: .github/skills/pytorch-cuda-xpu-triage/SKILL.md [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+- The change adds or modifies regression tests
+- The change touches `ATen/native` kernels, dispatch logic, or shared validation code
+- The bug involves edge-case semantics with a narrow trigger and clear signal
+- The fix lives in code paths likely inherited by XPU
+
+Reject when:
+- The change is infra-only, compiler-only, build/packaging/CI-only
+- Performance-only tuning with no regression signal or reproducer shape
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/SKILL.md]: This performance-only might still affect. Don't ignore it

File: .github/skills/pytorch-cuda-xpu-triage/SKILL.md [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+- The fix does not expose a plausible XPU parity gap
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/SKILL.md]: This also seems not correct. Prefer to delete it.

File: .github/skills/pytorch-cuda-xpu-triage/SKILL.md [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+### Step 2: Qualify candidates
+For each candidate, classify as **qualified** or **rejected**.
+
+Qualify when:
+- The change adds or modifies regression tests
+- The change touches `ATen/native` kernels, dispatch logic, or shared validation code
+- The bug involves edge-case semantics with a narrow trigger and clear signal
+- The fix lives in code paths likely inherited by XPU
+
+Reject when:
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/SKILL.md]: Do you have any analysis on what kind of "rejects" or "qualified" is in real cases?

For example, I scanned 30 days of commits, there are 1000 cases, of which, 100 cases are rejected because they are compiler-only, 200 cases are rejected because they are...

This might be more persuading.

Comment (laifenxiawucha) [reply .github/skills/pytorch-cuda-xpu-triage/SKILL.md]: update

File: .github/skills/pytorch-cuda-xpu-triage/references/github-mcp-reference.md (hunk 11 above)
Comment (laifenxiawucha) [reply .github/skills/pytorch-cuda-xpu-triage/references/github-mcp-reference.md]: update

File: .github/skills/pytorch-cuda-xpu-triage/references/github-mcp-reference.md
Comment (Stonepia) [inline .github/skills/pytorch-cuda-xpu-triage/references/github-mcp-reference.md:1]: This file name seems not correct. You are trying to github bug searching, instead of github-mcp-reference usage

Comment (Stonepia) [reply .github/skills/pytorch-cuda-xpu-triage/references/github-mcp-reference.md:1]: I almost think that this is not needed. This file does not provide any useful information

**Comment by Stonepia** [comment]
Overall seems ok for now. I have the concern about the pipeline:
1. Please specify in the PR description on what you are going to deliver (for example, what is the input of this skill and what is the output). We need an example to indicate what is the workflow of your skills.
2. The naming convention is not good. You are not actually doing the cuda-xpu-triage, you are just aligning the cuda fix for xpu on this skill, then give the reproducer. So this is not the issue triage, but an issue discovery task

**Review by Stonepia — APPROVED** [review]
Seems OK to me

File: .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md [hunk 21]
```diff
@@ -0,0 +1,186 @@
...
+Scan `pytorch/pytorch` for issues, PRs, and bug-fix commits across any backend (CUDA, ROCm, CPU, MPS) that may also affect XPU due to shared code paths. Adapt upstream reproducers for XPU, validate locally, route confirmed bugs, produce an auditable report.
+
+## Rules
+
+1. GitHub MCP first, `gh` CLI as fallback. Never hardcode tokens.
+2. Workspace-local XPU interpreter: `./.conda-xpu-triage/bin/python`
+3. Preflight checks nightly version; only reinstall if stale.
+4. `artifacts/candidate_ledger.jsonl` is the single source of truth.
+5. Zero pending actionable rows = done. Otherwise write `## Progress checkpoint`.
+6. Do not file issues or open PRs.
```
Comment (laifenxiawucha) [inline .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md]: 删除

File: .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md [hunk 22]
```diff
@@ -0,0 +1,186 @@
...
+---
+
+# PyTorch CUDA-Fix XPU Alignment
+
+Scan `pytorch/pytorch` for issues, PRs, and bug-fix commits across any backend (CUDA, ROCm, CPU, MPS) that may also affect XPU due to shared code paths. Adapt upstream reproducers for XPU, validate locally, route confirmed bugs, produce an auditable report.
+
+## Rules
+
+1. GitHub MCP first, `gh` CLI as fallback. Never hardcode tokens.
+2. Workspace-local XPU interpreter: `./.conda-xpu-triage/bin/python`
```
Comment (laifenxiawucha) [inline .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md]: 限制在 本工作区，没有可以重装，这个限制有点紧

File: .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md [hunk 23]
```diff
@@ -0,0 +1,186 @@
...
+3. Preflight checks nightly version; only reinstall if stale.
```
Comment (laifenxiawucha) [inline .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md]: add：保证最新版本

File: .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md [hunk 24]
```diff
@@ -0,0 +1,186 @@
...
+4. `artifacts/candidate_ledger.jsonl` is the single source of truth.
```
Comment (laifenxiawucha) [inline .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md]: 无用似乎

File: .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md [hunk 25]
```diff
@@ -0,0 +1,186 @@
...
+## Workflow
+
+### Step 0: Preflight
+
+```bash
+./.conda-xpu-triage/bin/python -c "import torch; print(torch.__version__); assert torch.xpu.is_available()"
+gh api repos/pytorch/pytorch --jq '.full_name'
+mkdir -p artifacts/details reports scripts
+./.conda-xpu-triage/bin/python -m torch.utils.collect_env > artifacts/collect_env.txt
+```
```
Comment (laifenxiawucha) [inline .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md]: 太过具体

File: .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md [hunk 26]
```diff
@@ -0,0 +1,186 @@
...
+Search `pytorch/pytorch` using GitHub MCP (fall back to `gh` CLI). Use caller-specified time window. Paginate with `per_page=100`; split date ranges if hitting the 1000-result cap.
+
+**Source types:**
+
+1. **Issues** — `repo:pytorch/pytorch is:issue` + bug-signal keywords (`crash`, `incorrect`, `wrong`, `regression`, `segfault`, `fail`)
+2. **PRs** — `repo:pytorch/pytorch is:pr is:merged` + fix-signal keywords (`fix`, `bug`, `correct`, `resolve`)
+3. **Commits** — search merged commits that fix backend bugs:
+   - `gh api search/commits?q=repo:pytorch/pytorch+fix+<keyword>+committer-date:>YYYY-MM-DD` (use commit search API)
+   - Or fetch commits from merged bug-fix PRs found in step (2)
+   - Look for commits touching `aten/`, `torch/`, `c10/` with fix-related messages
```
Comment (laifenxiawucha) [inline .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md]: 太具体

File: .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md [hunk 27]
```diff
@@ -0,0 +1,186 @@
...
+for f in scripts/repro_*.py; do
+  id=$(basename "$f" .py | sed 's/repro_//')
+  timeout 120 ./.conda-xpu-triage/bin/python "$f" > "artifacts/output_${id}.log" 2>&1
+  echo "$id exit=$?"
+done
+```
+
+Parse each `RESULT:` line → update ledger (`local_status: "done"`, `local_bucket`).
+
+If tensor `.device` is `cpu`, mark `blocked-script-error` (CPU fallback, not valid XPU test).
```
Comment (laifenxiawucha) [inline .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md]: 是否真的需要这个太过具体，是否需要依次进行

File: .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md [hunk 28]
```diff
@@ -0,0 +1,169 @@
...
+4. Zero pending actionable rows = done. Otherwise write `## Progress checkpoint`.
+5. **Batched pipeline**: batch deep-filter → batch write repros → serial execution → batch route → batch report. Ledger enables resume from any interruption.
+
+## Key Tables
+
+### CUDA-to-XPU Mapping
+
+| CUDA | XPU |
+|------|-----|
+| `torch.cuda.*` | `torch.xpu.*` |
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md]: Those things are not needed. The LLM is already trained and supposed to know that

Comment (Stonepia) [reply .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md]: I think we only add when the LLM doesn't work well

File: .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md [hunk 29]
```diff
@@ -0,0 +1,169 @@
...
+
+Verify XPU torch import and `torch.xpu.is_available()`, verify GitHub API access, create output directories (`artifacts/details`, `reports`, `scripts`), and save `collect_env` output.
+
+### Step 1: Collect candidates
+
+Search `pytorch/pytorch` using GitHub MCP (fall back to `gh` CLI). Use caller-specified time window. Paginate with `per_page=100`; split date ranges if hitting the 1000-result cap.
+
+**Principle**: cast a wide net — prefer over-collecting then filtering, rather than missing candidates at the search stage.
+
+**Source types:**
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md:74]: I preferer not to hard-code the things here. Do you have any proof that the labels covered all the issues? So I highly recommend that you rely on LLM to identify whether it is a bug or not

File: .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md [hunk 30]
```diff
@@ -0,0 +1,169 @@
...
+
+Generate `reports/full_scan.md` with all tested candidates:
+```
+1. `#<id>` — Title (kind: issue/pr/commit)
+Summary: ...
+Evidence: <url>
+Local XPU result: `<bucket>`
+```
+
+### Step 3: Audit
```
Comment (Stonepia) [inline .github/skills/pytorch-cuda-fix-xpu-alignment/SKILL.md:153]: Seems this step could be added into the issue-creation? we don't need the jsonl file ?
